frontend/src/metabase-lib/parameters/utils/targets.ts

```diff
@@ -52,10 +52,6 @@ export function getTargetFieldFromCard(
   card: Card,
   metadata: Metadata,
 ) {
-  if (!card?.dataset_query) {
-    return null;
-  }
```

Comment:
This was added in https://github.com/metabase/metabase/pull/20678, but the repro from that PR still works without this check `e2e/test/scenarios/dashboard-filters/reproductions/20656-dashboard-breaks-for-user-without-card-permissions.cy.spec.js`. I think this check is now outdated given we have improved our `metabase-lib` consistently.

I justify removing this check to simplify our code and remove further confusion.